src/AbstractSQLOptimiser.ts

```diff
@@ -915,23 +893,42 @@ const typeRules: Dictionary<MatchFn> = {
 					throw new SyntaxError(`'InsertQuery' does not support '${type}'`);
 			}
 		}
+		if (tables.length === 0) {
+			throw new SyntaxError("'InsertQuery' must have a From");
```

Comment:
This sounds a bit confusing -- I think it's better to change it like below to "InsertQuery requires a From component"

Reply:
Makes sense